shell/client/shell.html

```diff
@@ -487,6 +487,56 @@
   </div>
 </template>
 
+<template name="appdemo">
+  <div id="topbar">{{> homeLink}} App demo</div>
+
+  <div id="demo" class="main-content">
+    <div class="centered-box">
+      <h2></h2>
+      {{#if allowDemo}}
+        {{#if isSignedUp}}
+          {{#if isDemoUser}}
+            <p>You are currently logged in as a one-hour demo user.</p>
+            <p><button onclick="browseHome()">Continue the demo</button></p>
+          {{else}}
+            <p>You are already logged in as a real user.</p>
+          {{/if}}
+        {{else}}
+          {{#if loggingIn}}
+            <p>Logging in...</p>
+          {{else}}
+            <p><input id="demo-display-name" type="hidden" value="">
+               <button id="createDemoUser">Try {{ appName }} »</button></p>
+          {{/if}}
+        {{/if}}
+
+        <h3>What is this?</h3>
+        <p>If you click above, you'll open a temporary instance of {{
+          appName }} on the Sandstorm demo server.</p>
+
+	<p>Sandstorm is an open source hosting platform for personal
```

Comment:
FWIW, what I've done is turned the `demo` template into something more configurable, and have the `appdemo` template merely load in the entire `demo` template.